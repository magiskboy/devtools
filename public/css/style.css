.h-100 {
  height: 100%;
}

.h-75 {
  height: 75%;
}

.h-50 {
  height: 50%;
}

.h-25 {
  height: 25%;
}

.cm-gutters {
  background-color: var(--bulma-body-background-color) !important;
}