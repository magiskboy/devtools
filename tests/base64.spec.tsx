import React from 'react';
import { test, expect } from 'vitest';
import { getByTestId, render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { routeTree } from '../src/routeTree.gen';
import { createRouter, RouterProvider, createMemoryHistory } from '@tanstack/react-router';
import 'bulma/css/bulma.min.css';

test('base64', async () => {
  const router = createRouter({ routeTree });
  const history = createMemoryHistory({ initialEntries: ['/base64'] });
  const { container, getByText, getByPlaceholderText } = render(
    <RouterProvider router={router} history={history} />
  );

  // Wait for the route to be ready and component mounted
  await waitFor(() => {
    expect(container.querySelector('.cm-content')).toBeTruthy();
  });

  // Test initial render
  expect(container).toMatchSnapshot();
  
  // Test encode/decode buttons exist
  const encodeBtn = getByTestId(container, 'encode-button');
  const decodeBtn = getByTestId(container, 'decode-button');
  expect(encodeBtn).toBeDefined();
  expect(decodeBtn).toBeDefined();

  // Test input fields exist
  const dataInput = getByPlaceholderText('Enter text to encode/decode');
  expect(dataInput).toBeDefined();

  // Test auto-convert checkbox exists
  const autoConvertCheckbox = container.querySelector('input[type="checkbox"]');
  expect(autoConvertCheckbox).toBeDefined();
  if (!autoConvertCheckbox) throw new Error('Checkbox not found');

  // Test encoding
  const testText = 'Hello World';
  const encodedText = 'SGVsbG8gV29ybGQ=';
  
  await userEvent.type(dataInput, testText);
  await userEvent.click(encodeBtn);
  
  const encodedOutput = container.querySelectorAll('.cm-content')[1];
  expect(encodedOutput.textContent).toBe(encodedText);

  // Test decoding
  await userEvent.clear(dataInput);
  await userEvent.type(encodedOutput, encodedText);
  await userEvent.click(decodeBtn);
  
  const decodedOutput = container.querySelectorAll('.cm-content')[0];
  expect(decodedOutput.textContent).toBe(testText);

  // Test auto-convert
  await userEvent.click(autoConvertCheckbox);
  await userEvent.type(dataInput, 'Test');
  expect(encodedOutput.textContent).toBe('VGVzdA==');

  // Test error handling
  await userEvent.clear(encodedOutput);
  await userEvent.type(encodedOutput, 'invalid base64!');
  await userEvent.click(decodeBtn);
  expect(getByText('Invalid Base64 format')).toBeDefined();
})
