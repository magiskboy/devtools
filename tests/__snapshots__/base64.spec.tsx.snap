// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`base64 1`] = `
<div>
  <div
    class="container is-flex is-flex-direction-column h-100"
  >
    <nav
      class="navbar is-transparent mb-5"
    >
      <div
        class="navbar-brand"
      >
        <a
          class="navbar-item"
          href="/"
        >
          <img
            height="32"
            src="/favicon.png"
            width="32"
          />
        </a>
        <div
          class="navbar-burger js-burger"
          data-target="devtoolsNavbar"
        >
          <span />
          <span />
          <span />
          <span />
        </div>
      </div>
      <div
        class="navbar-menu"
        id="devtoolsNavbar"
      >
        <div
          class="navbar-start"
        >
          <a
            class="navbar-item"
            href="/"
          >
             
            Base64
             
          </a>
        </div>
        <div
          class="navbar-end"
        >
          <div
            class="navbar-item"
          >
            <div
              class="field is-grouped"
            >
              <p
                class="control"
              >
                <a
                  href="https://github.com/magiskboy/devtools"
                  style="display: block;"
                  target="_blank"
                >
                  <span>
                     Github 
                  </span>
                </a>
              </p>
            </div>
          </div>
          <div
            class="navbar-item"
          >
            <div
              class="field is-grouped"
            >
              <p
                class="control"
              >
                <a
                  href="https://nkthanh.dev"
                  target="_blank"
                >
                  <span>
                     About me 
                  </span>
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="fixed-grid has-9-cols"
    >
      <div
        class="grid"
      >
        <div
          class="cell is-col-span-4"
        >
          <div
            class="_container_geklt_1"
          >
            <div
              class="_header_geklt_6"
            >
              <p
                class="_title_geklt_14"
              >
                Data
              </p>
              <button
                aria-label="Copy code"
                class="_copyButton_geklt_21 "
              >
                Copy
              </button>
            </div>
            <div
              class="cm-theme-light _content_geklt_39 "
              style="width: 148px;"
            >
              <div
                class="cm-editor ͼ1 ͼ2 ͼ4 ͼ18 ͼ15"
                style="background-color: transparent;"
              >
                <div
                  aria-live="polite"
                  class="cm-announced"
                />
                <div
                  class="cm-scroller"
                  tabindex="-1"
                >
                  <div
                    aria-hidden="true"
                    class="cm-gutters"
                    style="min-height: 26.1953px; position: sticky;"
                  >
                    <div
                      class="cm-gutter cm-lineNumbers"
                    >
                      <div
                        class="cm-gutterElement"
                        style="height: 0px; visibility: hidden; pointer-events: none;"
                      >
                        9
                      </div>
                      <div
                        class="cm-gutterElement cm-activeLineGutter"
                        style="height: 18.1953px; margin-top: 4px;"
                      >
                        1
                      </div>
                    </div>
                    <div
                      class="cm-gutter cm-foldGutter"
                    >
                      <div
                        class="cm-gutterElement"
                        style="height: 0px; visibility: hidden; pointer-events: none;"
                      >
                        <span
                          title="Unfold line"
                        >
                          ›
                        </span>
                      </div>
                      <div
                        class="cm-gutterElement cm-activeLineGutter"
                        style="height: 18.1953px; margin-top: 4px;"
                      />
                    </div>
                  </div>
                  <div
                    aria-multiline="true"
                    autocapitalize="off"
                    autocorrect="off"
                    class="cm-content"
                    contenteditable="true"
                    role="textbox"
                    spellcheck="false"
                    style="tab-size: 4;"
                    translate="no"
                    writingsuggestions="false"
                  >
                    <div
                      class="cm-activeLine cm-line"
                    >
                      <img
                        aria-hidden="true"
                        class="cm-widgetBuffer"
                      />
                      <span
                        aria-label="placeholder Enter text to encode/decode"
                        class="cm-placeholder"
                        contenteditable="false"
                        style="pointer-events: none;"
                      >
                        Enter text to encode/decode
                      </span>
                      <br />
                    </div>
                  </div>
                  <div
                    aria-hidden="true"
                    class="cm-layer cm-layer-above cm-cursorLayer"
                    style="z-index: 150; animation-duration: 1200ms;"
                  >
                    <div
                      class="cm-cursor cm-cursor-primary"
                      style="left: 36.8281px; top: 5.5px; height: 15px;"
                    />
                  </div>
                  <div
                    aria-hidden="true"
                    class="cm-layer cm-selectionLayer"
                    style="z-index: -2;"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="cell is-flex is-flex-direction-column is-justify-content-center is-align-items-center"
        >
          <button
            class="button is-primary my-4"
            data-testid="encode-button"
            disabled=""
          >
            Encode
          </button>
          <button
            class="button is-secondary"
            data-testid="decode-button"
            disabled=""
          >
            Decode
          </button>
          <label
            class="checkbox mt-4"
          >
            <input
              type="checkbox"
            />
            Auto-convert
          </label>
        </div>
        <div
          class="cell is-col-span-4"
        >
          <div
            class="_container_geklt_1"
          >
            <div
              class="_header_geklt_6"
            >
              <p
                class="_title_geklt_14"
              >
                Base64
              </p>
              <button
                aria-label="Copy code"
                class="_copyButton_geklt_21 "
              >
                Copy
              </button>
            </div>
            <div
              class="cm-theme-light _content_geklt_39 "
              style="width: 148px;"
            >
              <div
                class="cm-editor ͼ1 ͼ2 ͼ4 ͼ19 ͼ15"
                style="background-color: transparent;"
              >
                <div
                  aria-live="polite"
                  class="cm-announced"
                />
                <div
                  class="cm-scroller"
                  tabindex="-1"
                >
                  <div
                    aria-hidden="true"
                    class="cm-gutters"
                    style="min-height: 26.1953px; position: sticky;"
                  >
                    <div
                      class="cm-gutter cm-lineNumbers"
                    >
                      <div
                        class="cm-gutterElement"
                        style="height: 0px; visibility: hidden; pointer-events: none;"
                      >
                        9
                      </div>
                      <div
                        class="cm-gutterElement cm-activeLineGutter"
                        style="height: 18.1953px; margin-top: 4px;"
                      >
                        1
                      </div>
                    </div>
                    <div
                      class="cm-gutter cm-foldGutter"
                    >
                      <div
                        class="cm-gutterElement"
                        style="height: 0px; visibility: hidden; pointer-events: none;"
                      >
                        <span
                          title="Unfold line"
                        >
                          ›
                        </span>
                      </div>
                      <div
                        class="cm-gutterElement cm-activeLineGutter"
                        style="height: 18.1953px; margin-top: 4px;"
                      />
                    </div>
                  </div>
                  <div
                    aria-multiline="true"
                    autocapitalize="off"
                    autocorrect="off"
                    class="cm-content"
                    contenteditable="true"
                    role="textbox"
                    spellcheck="false"
                    style="tab-size: 4;"
                    translate="no"
                    writingsuggestions="false"
                  >
                    <div
                      class="cm-activeLine cm-line"
                    >
                      <img
                        aria-hidden="true"
                        class="cm-widgetBuffer"
                      />
                      <span
                        aria-label="placeholder Enter Base64 string to decode"
                        class="cm-placeholder"
                        contenteditable="false"
                        style="pointer-events: none;"
                      >
                        Enter Base64 string to decode
                      </span>
                      <br />
                    </div>
                  </div>
                  <div
                    aria-hidden="true"
                    class="cm-layer cm-layer-above cm-cursorLayer"
                    style="z-index: 150; animation-duration: 1200ms;"
                  >
                    <div
                      class="cm-cursor cm-cursor-primary"
                      style="left: 36.8281px; top: 5.5px; height: 15px;"
                    />
                  </div>
                  <div
                    aria-hidden="true"
                    class="cm-layer cm-selectionLayer"
                    style="z-index: -2;"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="modal"
      style="justify-content: initial;"
    >
      <div
        class="modal-background"
      />
      <div
        class="modal-content"
        style="position: absolute; margin-top: 10%;"
      >
        <div
          class="box"
          style="min-height: 400px;"
        >
          <div
            class="control"
          >
            <input
              class="input"
              placeholder="Search..."
              type="text"
              value=""
            />
          </div>
          <div
            class="mt-4"
          >
            <div
              class="box"
            >
              <a
                href="/jsonnet"
              >
                Jsonnet
              </a>
            </div>
            <div
              class="box"
            >
              <a
                href="/jwt-decode"
              >
                JWT decode
              </a>
            </div>
            <div
              class="box"
            >
              <a
                href="/sql-fmt"
              >
                SQL fmt
              </a>
            </div>
            <div
              class="box"
            >
              <a
                href="/yaml-json"
              >
                YAML - JSON
              </a>
            </div>
            <div
              class="box"
            >
              <a
                href="/yaml-fmt"
              >
                YAML fmt
              </a>
            </div>
          </div>
        </div>
      </div>
      <button
        class="modal-close is-large"
      />
    </div>
    <div
      class="py-4"
    >
      <footer
        class="_footer_1d5za_1"
      >
        <div
          class="_content_1d5za_7"
        >
          <p>
            My IP: 
            117.5.110.214
          </p>
          <p>
            All data and HTTP requests to fetch JSON are processed locally in your browser.
          </p>
          <p>
            Copyright © 
            2025
             -
             
            <a
              aria-label="Visit nkthanh.dev (opens in new tab)"
              class="_link_1d5za_24"
              href="https://nkthanh.dev"
              rel="noopener noreferrer"
              target="_blank"
            >
              nkthanh.dev
            </a>
          </p>
        </div>
      </footer>
    </div>
  </div>
</div>
`;
